src/array/cuda/spmm.cu

```diff
@@ -282,6 +282,107 @@ void CusparseCsrmm2(
   if (valptr)
     device->FreeWorkspace(ctx, valptr);
 }
+
+/*! Cusparse implementation of SpMM on Csr format. */
+template <typename DType, typename IdType>
+void CusparseCsrmm2Hetero(
+    const DLContext& ctx,
+    const CSRMatrix& csr,
+    const DType* B_data, const DType* A_data,
+    DType* C_data,
+    int x_length,
+    cudaStream_t strm_id) {
```

Comment:
Since with the current implementation, this function will only work correctly if this is the default stream, I think it would be good to add something like:
```
CHECK_EQ(strm_id, 0) << "CusparseCsrmm2Hetero is only implemented for the default stream currently.";
```